rfd/0191-workload-id-config-ux.md

```diff
@@ -0,0 +1,1467 @@
+    jwt:
+      sub: spiffe://example.teleport.sh/gitlab/my-org/my-project/production
+      iss: https://example.teleport.sh
+      -- snipped for conciseness: all JWT claims would be included --
+    x509:
+      subject:
+        common_name: spiffe://example.teleport.sh/gitlab/my-org/my-project/production
+        organization: my-org
+        -- snipped for conciseness: all Subject DN fields would be included --
+      sans:
+      - dns: my-project.example.com
+      - ip: 10.0.0.1
+      - uri: spiffe://example.teleport.sh/gitlab/my-org/my-project/production 
+2 WorkloadIdentity resources did not match the given attributes:
+
+- workload_identity_name: gitlab-staging
+  reason: Rule evaluation failed. Allow rule expression `join.gitlab.environment == "production"` returned false.
+- workload_identity_name: github-production
+  reason: The `join.github.environment` attribute included in `spec.spiffe.id` template do not exist in the attribute set.
+```
+
+Similar functionality should be built into the Teleport Web UI. This should
+initially be built into the WorkloadIdentity resource view, and allow the
+resource to be tested against a set of attributes derived from an existing
+Bot Instance resource.
+
+Later, this functionality can be built into the "Create or Edit" flow for a 
+WorkloadIdentity resource.
+
+### X509 SVIDs and DNS SANs
+
+By default, the X509 SVIDs issued for a WorkloadIdentity resource will not 
+include any DNS SANs.
+
+To include DNS SANs, the `spec.spiffe.x509.dns_sans` field can be used:
+
+```yaml
+kind: workload_identity
+version: v1
+metadata:
+  name: gitlab
+spec:
+  spiffe:
+    id: /my-awesome-service
+    x509:
+      dns_sans:
+      - my.awesome.service.example.com
+      - "*.my.awesome.service.example.com"
+```
+
+This will result in any X509 SVID issued using this WorkloadIdentity including
+both the specified DNS SANs.
+
+As with `spec.spiffe.id`, templating can be used to include attributes within
+DNS SANs:
+
+```yaml
+kind: workload_identity
+version: v1
+metadata:
+  name: gitlab
+spec:
+  spiffe:
+    id: /my-awesome-service
+    x509:
+      dns_sans:
+      - {{ join.gitlab.environment }}.gitlab.example.com 
+```
+
+### Time To Live (TTL)
+
+Both X509 and JWT SVIDs include fields which control when the credential is
+valid from and when it is valid to.
+
+When requesting the issuance of a credential, the `tbot` agent will provide a
+requested TTL for the credential - which will either be a default value or
+one explicitly configured by a user.
+
+However, the WorkloadIdentity resource will also include a `spec.spiffe.ttl.max`
+field. If `tbot` requests a credential with a TTL greater than this value, the
+TTL will be capped at this value. This provides the ability to enforce a 
+maximum permissible TTL regardless of the configuration of the `tbot` agent.
+
+If this field is not set, a default maximum value of 24 hours will be used.
```

Comment:
Do we / the SPIFFE spec define an upper limit to the TTL, or is it arbitrary?

Reply:
SPIFFE doesn't specify anything. Across the rest of Teleport we do have some limits, but, I'm leaning towards that we may want to relax these for Workload Identity in order to support some more niche use-cases.